include/libdnf/module/module_item.hpp

```diff
@@ -138,13 +140,15 @@ class ModuleItem {
     // @replaces libdnf:module/ModuleItem.hpp:method:ModuleItem.getStaticContext()
     bool get_static_context() const;
 
-    /// @return The list of strings of modules requires (e.g. "nodejs", "nodejs:12", "nodejs:-11").
+    std::vector<ModuleDependencies> get_module_dependencies() const;
```

Comment:
I was thinking whether is it correct or nor and I can confirm that it is according to libmodulemd specifications.
Dependencies contain `std::vector<std::map<string, vector<string>>>`.

I have let say less correct proposal. At the end we use all elements (name and stream) with the same impact and we do not care whether they were in the same map or a different one. Therefore we can simplify the structure to `std::vector<std::pair<std::string, std::vector<std::string>>>`. But `std::pair<std::string, std::vector<std::string>>` is not nice and not transparent therefore we can replace it by class Require or Dependency that will contain stream name (string) and streams (sorted vector with strings). I can imagine that it will have 3 methods - get_name, get_streams, and to_string. Because it will require a default public constructor for binding I recommend when someone will call get_name() and inside will be an empty string I suppose to raise a logic error or a libdnf assert.  As I said it is not a correct representation but may be it can simplify the usage of dependencies. What do you think?

Reply:
Hi, thanks for your suggestion, I changed it to the vector of ModuleDependency objects.